docs/content/develop/step-2e.md

```diff
@@ -0,0 +1,319 @@
+
+Let's dig into the schema for our pfe-cool-element component:
+
+```
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Cool element",
+  "description": "This element has a profile picture and a follow button.",
+  "type": "object",
+  "tag": "pfe-cool-element",
+  "class": "PfeCoolElement",
+  "category": "content",
+  "properties": {
+    "slots": {
+      "title": "Slots",
+      "description": "Definition of the supported slots",
+      "type": "object",
+      "properties": {
+        "default": {
+          "title": "Default",
+          "type": "array",
+          "namedSlot": false,
+          "items": {
+            "oneOf": [
+              {
+                "$ref": "raw"
+              }
+            ]
+          }
+        }
+      }
+    },
+    "attributes": {
+      "title": "Attributes",
+      "type": "object",
+      "properties": {
+        "following": {
+          "title": "Follow this user",
+          "type": "boolean",
+          "default": false,
+          "prefixed": true,
+          "observer": "_followToggle"
+        },
+        "photo-url": {
+          "title": "Profile image",
+          "type": "string",
+          "observer": "_addImage"
+        }
+      }
+    },
+    "events": {
+      "title": "Events",
+      "description": "Definition of the supported events",
+      "type": "object",
+      "properties": {
+        "follow": {
+          "title": "Follow",
+          "type": "string",
+          "default": "pfe-cool-element:follow"
+        }
+      }
+    }
+  },
+  "required": ["slots", "attributes", "events"],
+  "additionalProperties": false
+}
+```
+
+## General information
+
+At the top of our schema is general information about our component: title, description, the tag name and class, and the category of component it falls under.  We also see the version of the schema draft we are using here.
+
+```
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Cool element",
+  "description": "This element has a profile picture and a follow button.",
+  "type": "object",
+  "tag": "pfe-cool-element",
+  "class": "PfeCoolElement",
+  "category": "content",
+```
+
+A schema is a JSON object with certain protected keys: `title`, `description`, `type`, and `required` are top among these.  For our web component, we specify the slots, attributes, and events as nested objects in the `properties` object of the overall component.
+
+## Slots 
+
+We list out our slots in the properties object of the schema.  Every slot is an array type and can contain either specific tags or a generic `raw` reference which means any HTML can be accepted here.  If the slot in question is a named slot such as `<slot name="pfe-cool-element--region">`, change the `"namedSlot"` boolean to true.
```

Comment:
re:Kendall

Just curious, are we using the name "raw" as a holdover from webrh? Does it make more sense to call it "any" or "any_html_element" ..? Is it way too late to change this? :)

Reply:
We're not really using this at all.  I'm open to other ideas though not sure `any` is specific enough and `any_html_element` is a weird syntax.  Maybe any HTML just leaves this section blank with no requirements and if it's an opinionated slot, then it lists the components it can accept.